addons/mail/static/src/components/notification_message_view/notification_message_view.js

```diff
@@ -1,11 +1,47 @@
 /** @odoo-module **/
 
+import { useComponentToModel } from '@mail/component_hooks/use_component_to_model';
+import { useUpdateToModel } from '@mail/component_hooks/use_update_to_model';
 import { registerMessagingComponent } from '@mail/utils/messaging_component';
 
 const { Component } = owl;
 
 class NotificationMessageView extends Component {
 
+    /**
+     * @override
+     */
+     setup() {
+        super.setup();
+        useComponentToModel({ fieldName: 'component' });
+        useUpdateToModel({ methodName: 'onComponentUpdate' });
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Tell whether the notification message is partially visible on browser window or not.
+     *
+     * @returns {boolean}
+     */
+    isPartiallyVisible() {
+        if (!this.root.el) {
+            return false;
+        }
+        const elRect = this.root.el.getBoundingClientRect();
+        if (!this.root.el.parentNode) {
+            return false;
+        }
+        const parentRect = this.root.el.parentNode.getBoundingClientRect();
+        // intersection with 5px offset
+        return (
+            elRect.top < parentRect.bottom + 5 &&
+            parentRect.top < elRect.bottom + 5
+        );
```

Comment:
This is duplicate of `MessageView/isPartiallyVisible`.
I think you could just move this code in `MessageListViewItem` so that both kind of items use the same logic.

Reply:
I updated :)